components/google_calendar/google_calendar.app.js

```diff
@@ -1,154 +1,307 @@
-const { google } = require("googleapis")
-const get = require("lodash.get")
+const { google } = require("googleapis");
+const get = require("lodash.get");
 
 module.exports = {
   type: "app",
   app: "google_calendar",
   propDefinitions: {
     calendarId: {
+      label: "Calender Name",
       description: "Calendar identifier. To retrieve calendar IDs call the calendarList.list method. If you want to access the primary calendar of the currently logged in user, use the \"primary\" keyword.",
-      type: "string"
+      type: "string",
+      optional: true,
+      default: "primary",
+      async options({ prevContext }) {
+        const { nextPageToken } = prevContext;
+        const calendars = await this.listCalendarsInPage(nextPageToken);
+        const options = [];
+        for (const calendar of calendars.items) {
+          options.push({
+            label: calendar.summary,
+            value: calendar.id,
+          });
+        }
+        return {
+          options,
+          context: {
+            nextPageToken,
+          },
+        };
+      },
```

Comment:
I think you are not handling pagination the right way. This is my suggestion
```suggestion
     async options({
        page, prevContext,
      }) {
        const { nextPageToken } = prevContext;

        if (page && !nextPageToken) {
          return [];
        }

        const data =
          await this.listCalendars({
            pageToken: nextPageToken,
            showHidden: true,
          });

        const { items: calendars } = data;

        const options =
          calendars.map(({
            id,
            summary,
          }) => ({
            label: summary,
            value: id,
          }));

        return {
          options,
          context: {
            nextPageToken: data.nextPageToken,
          },
        };
      },
    },
```

In that way you want to refactor the function `calendarList` to
```javascript
    async listCalendars(params) {
      const calendar = this.calendar();
      const { data } =  await calendar.calendarList.list(params);
      return data;
    },
```

and then you no longer need the function `listCalendarsInPage` so you can remove it